smtp/response.js

```diff
@@ -48,46 +66,74 @@ class SMTPResponse {
 			);
 		};
 
+		/**
+		 * @param {string | Buffer} data the data
+		 * @returns {void}
+		 */
 		const watch = data => {
 			if (data !== null) {
 				buffer += data.toString();
 				notify();
 			}
 		};
 
+		/**
+		 * @param {Error} err the error object
+		 * @returns {void}
+		 */
 		const close = err => {
 			stream.emit(
 				'response',
 				SMTPError('connection has closed', SMTPError.CONNECTIONCLOSED, err)
 			);
 		};
 
+		/**
+		 * @param {Error} err the error object
+		 * @returns {void}
+		 */
 		const end = err => {
 			stream.emit(
 				'response',
 				SMTPError('connection has ended', SMTPError.CONNECTIONENDED, err)
 			);
 		};
 
+		/**
+		 * @param {Error} [err] the error object
+		 * @returns {void}
+		 */
 		this.stop = err => {
 			stream.removeAllListeners('response');
-			stream.removeListener('data', watch);
-			stream.removeListener('end', end);
-			stream.removeListener('close', close);
-			stream.removeListener('error', error);
+			if (stream instanceof TLSSocket) {
```

Comment:
according to types, would throw if it's a `Socket`

Reply:
when would it be a Socket instead of a TLSSocket?

Reply:
inside `[./smtp/smtp.js](https://github.com/zackschuster/emailjs/blob/drop-starttls/smtp/smtp.js#L323)` - `this.sock` can be either `Socket` or `TLSSocket` based on the immediately preceding code

Reply:
this was mistyped; `NodeJS.Socket` isn't the same as `import('net').Socket`. it's fine to call these methods on either.